### DISCUSSION TITLE
FastAPI-Users v9 cookies handling in tests

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#838] [Q&A] [answered]
While [updating Jupyverse with FastAPI-Users v9](https://github.com/jupyter-server/jupyverse/pull/142) I noticed that some changes are required in the tests. I'm not sure it's because of v9, but now the cookies are set in the `TestClient` as a [RequestsCookieJar](https://docs.python-requests.org/en/master/api/#requests.cookies.RequestsCookieJar), and I had to [access/pass cookies differently](https://github.com/jupyter-server/jupyverse/pull/142/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128), e.g.:
```python
response = client.get("/auth/user/me", cookies=client.cookies.get_dict())
```
While with v8 I could just do:
```python
response = client.get("/auth/user/me")
```
Does it ring a bell by any chance?

**Comment by frankie567** [comment, marked as answer]
Hmm, I think it's because you forgot to pass cookie parameters. By default, cookies are HTTPS only; so they are not kept in the test client context.

## What you had before

```py
cookie_authentication = CookieAuthentication(
    secret=secret, cookie_secure=get_auth_config().cookie_secure, name="cookie"  # type: ignore
```

## What you have now

```py
cookie_transport = CookieTransport()
```

## What you *should* have

```py
cookie_transport = CookieTransport(cookie_secure=get_auth_config().cookie_secure)
```

This way, you can have `cookie_secure=False` during testing (and avoid the need to pass the cookies manually) and `cookie_secure=True` in production.

**Reply by davidbrochart** [reply]
Wow, thanks a lot for spotting it!
